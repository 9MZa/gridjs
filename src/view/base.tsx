import { Component } from 'preact';

export interface BaseProps {
  classNamePrefix?: string
}

export abstract class BaseComponent<P extends BaseProps, S> extends Component<P, S> {}
