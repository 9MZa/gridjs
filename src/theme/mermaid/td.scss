@import "colors";

td.gridjs {
  &-td {
    border: 1px solid $gray3;
    padding: 12px 24px;
    background-color: $white;
  }

  &-td:first-child {
    border-left: none;
  }

  &-td:last-child {
    border-right: none;
  }
}
