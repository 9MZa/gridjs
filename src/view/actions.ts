import Tabular from '../tabular';
import { Status } from '../types';

export const SetStatusToRendered = () => (state) => {
  if (state.status === Status.Loaded) {
    return SetStatus(Status.Rendered);
  }

  return state;
};

export const SetStatus = (status: Status) => (state) => {
  return {
    ...state,
    status: status,
  };
};

export const SetData = (data: Tabular) => (state) => {
  if (!data) return state;

  return {
    ...state,
    data: data,
    status: Status.Loaded,
  };
};

export const SetNoData = () => (state) => {
  return {
    ...state,
    data: null,
    status: Status.Error,
  };
};
