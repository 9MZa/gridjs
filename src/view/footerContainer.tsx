import { h } from 'preact';

import { BaseComponent } from './base';
import { className } from '../util/className';
import { Pagination } from './plugin/pagination';
import { useRef } from 'preact/hooks';
import getConfig from '../util/getConfig';

interface FooterContainerState {
  isActive: boolean;
}

export class FooterContainer extends BaseComponent<{}, FooterContainerState> {
  private footerRef = useRef(null);

  constructor(props, context) {
    super(props, context);

    this.state = {
      isActive: true,
    };
  }

  componentDidMount() {
    if (this.footerRef.current.children.length === 0) {
      this.setState({
        isActive: false,
      });
    }
  }

  render() {
    const config = getConfig(this.context);

    if (this.state.isActive) {
      return (
        <div ref={this.footerRef} className={className('footer')}>
          <Pagination {...config.pagination} />
        </div>
      );
    }

    return null;
  }
}
