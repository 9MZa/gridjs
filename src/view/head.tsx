import { h } from 'preact';

import Config from '../config';
import { BaseComponent } from './base';
import className from '../util/className';
import { Search } from './plugin/search';

import '../theme/mermaid/head.scss';

export class Head extends BaseComponent<{}, {}> {
  render() {
    return (
      <div className={className(Config.current.classNamePrefix, 'head')}>
        <Search />
      </div>
    );
  }
}
