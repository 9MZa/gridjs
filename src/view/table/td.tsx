import { h } from 'preact';

import Cell from '../../cell';
import { BaseComponent, BaseProps } from '../base';
import className from '../../util/className';
import { TCell } from '../../types';

export interface TDProps extends BaseProps {
  cell: Cell<TCell>;
  colSpan?: number;
  className?: string;
}

export class TD extends BaseComponent<TDProps, {}> {
  render() {
    return (
      <td
        colSpan={this.props.colSpan}
        className={`${className('td')}${
          this.props.className ? ' ' + this.props.className : ''
        }`}
      >
        {this.props.cell.data}
      </td>
    );
  }
}
