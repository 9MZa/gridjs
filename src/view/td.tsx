import { h } from 'preact';

import Cell from '../cell';
import {BaseComponent, BaseProps} from './base';
import className from "../util/className";
import Config from "../config";

import "../theme/mermaid/td.scss";

export interface TDProps extends BaseProps {
  cell: Cell;
}

export class TD extends BaseComponent<TDProps, {}> {
  render() {
    return <td className={className(Config.current.classNamePrefix, "td")}>
      {this.props.cell.data}
    </td>;
  }
}
