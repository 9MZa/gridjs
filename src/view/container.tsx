import { h } from 'preact';

import Tabular from '../tabular';
import Config from '../config';
import { BaseComponent, BaseProps } from './base';
import { Table } from './table';
import { Status, TBodyCell } from '../types';
import Header from '../header';
import className from '../util/className';

import '../theme/mermaid/container.scss';

interface ContainerProps extends BaseProps {
  config: Config;
}

interface ContainerState {
  status: Status;
  data?: Tabular<TBodyCell>;
  header?: Header;
}

export class Container extends BaseComponent<ContainerProps, ContainerState> {
  private readonly config: Config;

  constructor(props) {
    super(props);

    this.state = {
      status: Status.Init,
      data: null,
      header: this.props.config.header,
    };

    this.config = this.props.config;
  }

  componentWillMount(): void {
    this.setState({
      status: Status.Loading,
    });
  }

  async componentDidMount() {
    this.setState({
      data: await this.config.pipeline.process(),
      status: Status.Loaded,
    });
  }

  render() {
    return (
      <div className={className(Config.current.classNamePrefix, 'container')}>
        <Table
          data={this.state.data}
          header={this.state.header}
          classNamePrefix={this.config.classNamePrefix}
        />
      </div>
    );
  }
}
