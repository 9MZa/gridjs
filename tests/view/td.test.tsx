import React from "react";
import { TD } from '../../src/view/td';
import renderer from 'react-test-renderer';
import Cell from "../../src/cell";


describe('TD component', () => {
  it('should render a tr element', () => {
    const td = renderer.create(
      <TD cell={new Cell('boo')} />
    );
    expect(td).toMatchSnapshot();
  });
});
