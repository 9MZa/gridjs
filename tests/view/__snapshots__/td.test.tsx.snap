// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TD component should render a tr element 1`] = `
<td>
  boo
</td>
`;
